### PR TITLE
[circle] Use Docker image from Gitlab registry.

### PR DESCRIPTION
**Pull request by ejgallego**

### CODE REVIEW

**Comment by Zimmi48** [comment]
Maybe you can update the README and remove the push hook at the same time?

**Comment by ejgallego** [comment]
> Maybe you can update the README and remove the push hook at the same time?

Indeed it does make sense.

File: .circleci/config.yml [hunk 1]
```diff
@@ -46,7 +47,9 @@ before_script: &before_script
         paths:
           - coq/
 
-  environment: *envvars
+  environment:
+    <<: *envvars
+    NATIVE_COMP: "yes"
```
Comment (SkySkimmer) [inline .circleci/config.yml]: Why this change?

Comment (ejgallego) [reply .circleci/config.yml]: Otherwise the image vars are not picked up in subjobs.

File: dev/ci/docker/README.md [hunk 2]
```diff
@@ -1,37 +1,13 @@
 ## Overall Docker Setup for Coq's CI.
```
Comment (SkySkimmer) [inline dev/ci/docker/README.md]: We should say somewhere "if you change the dockerfile you need to change the cachekey in gitlab and circle yml files"

**Review by Zimmi48 — COMMENTED** [review]
You could also remove the three lines about `coqci` since it seems we're not going to use this organization after all.

File: dev/ci/docker/README.md (hunk 2 above)
Comment (ejgallego) [reply dev/ci/docker/README.md]: I think now things are simple enough that even a commit check could work to ensure this.

**Comment by ejgallego** [comment]
Sure, will do another pass.

File: dev/ci/docker/README.md (hunk 2 above)
Comment (Zimmi48) [reply dev/ci/docker/README.md]: That would be great!

**Comment by ejgallego** [comment]
Ok, changes done. I don't know what the best way to ensure automatically these 3:
```
$ git grep 'CACHEKEY:'
.circleci/config.yml:    CACHEKEY: "bionic_coq-V2018-05-07-V2"
.gitlab-ci.yml:  CACHEKEY: "bionic_coq-V2018-05-07-V2"
dev/ci/docker/bionic_coq/Dockerfile:# CACHEKEY: "bionic_coq-V2018-05-07-V2"
```
are kept in sync. I won't have the time to work on it soon so feel free to tackle on the pre-commit check.

File: dev/ci/docker/bionic_coq/Dockerfile [hunk 3]
```diff
@@ -1,3 +1,6 @@
+# CACHEKEY: "bionic_coq-V2018-05-07-V2"
```
Comment (Zimmi48) [inline dev/ci/docker/bionic_coq/Dockerfile]: What is the point of repeating the CACHEKEY here? Unless this isn't actually a real comment...

Comment (ejgallego) [reply dev/ci/docker/bionic_coq/Dockerfile]: The point is to allow a script to check that this version here is in sync with what's inside `.gitlab-ci.yml`

Comment (SkySkimmer) [reply dev/ci/docker/bionic_coq/Dockerfile]: For automatic checking we should be able to check that any commit modifying the dockerfile also modifies the cachekey in gitlab.yml.

Comment (ejgallego) [reply dev/ci/docker/bionic_coq/Dockerfile]: That's also a valid way to do it I guess. Also cool would be to autogenerate `.gitlab-ci.yml` and `Dockerfile` using a promote rule.

**Comment by SkySkimmer** [comment]
I have no clue what that is.

Gaëtan Gilbert

**Comment by ejgallego** [comment]
> I have no clue what that is.

I have no clude what "what" is but I infer you talk about promote rules.

A promote make rule is a build rule that allows to check consistent version of auto-generated files in the repository, kind of releasing a file. Thus, for example here, `.gitlab-ci.yml` could depend on the hash of the Dockerfile, so when you want to update it you just do "promote".

The build rule can also warn you when the file is out of date [as it will generate a temp version and then diff].

**Review by SkySkimmer — APPROVED** [review]
